django/contrib/auth/hashers.py

```diff
@@ -416,35 +442,52 @@ def encode(self, password, salt):
         data = bcrypt.hashpw(password, salt)
         return "%s$%s" % (self.algorithm, data.decode('ascii'))
 
-    def verify(self, password, encoded):
-        algorithm, data = encoded.split('$', 1)
+    def decode(self, encoded):
+        algorithm, empty, algostr, work_factor, data = encoded.split('$', 4)
         assert algorithm == self.algorithm
+        salt, checksum = data[:22], data[22:]
+        return {
+            'algorithm': algorithm,
+            'algostr': algostr,
+            'work factor': work_factor,
```

Comment:
I'd call this `work_factor` as dictionary key and only in `safe_summary` it would be `work_factor`

Reply:
Updated and fixed.